### PR TITLE
Improving site permissions API.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by Amejia481** [comment]
This also will fix: https://github.com/mozilla-mobile/fenix/issues/1416 and https://github.com/mozilla-mobile/fenix/issues/1557

File: components/feature/sitepermissions/src/main/res/layout/mozac_site_permissions_prompt.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            android:paddingStart="5dp"
+            android:paddingEnd="5dp"
+            android:layout_alignParentTop="true"
+            android:layout_marginTop="16dp"
+            android:layout_toEndOf="@id/icon"
+            android:layout_alignBaseline="@id/icon"
+            tools:text="Allow wikipedia.org to use your camera?"/>
+
+    <CheckBox
+            android:id="@+id/do_not_ask_again"
```
Comment (jonalmeida) [inline components/feature/sitepermissions/src/main/res/layout/mozac_site_permissions_prompt.xml]: We might need text wrapping on this line or padding to make it fit in the dialog. CoordinatorLayout fixes this too! 🙃 

Screen Shot 2019-05-06 at 4 00 07 PM

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+private const val KEY_POSITIVE_BUTTON_TEXT_COLOR = "KEY_POSITIVE_BUTTON_TEXT_COLOR"
+private const val KEY_SHOULD_INCLUDE_CHECKBOX = "KEY_SHOULD_INCLUDE_CHECKBOX"
+private const val KEY_IS_NOTIFICATION_REQUEST = "KEY_IS_NOTIFICATION_REQUEST"
+private const val DEFAULT_VALUE = Int.MAX_VALUE
+
+internal class SitePermissionsDialogFragment : AppCompatDialogFragment() {
+
+    internal var feature: SitePermissionsFeature? = null
+
+    internal val sessionId: String by lazy { safeArguments.getString(KEY_SESSION_ID) }
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragment.kt]: It was a nice way to just get the value once and wrap the long call in field.

[the remark this answers is not in the document]
Comment (Amejia481) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragment.kt]: I was taking a look to some notes from [this talk at Android makers (Kotlin Under the Covers)](https://androidmakers.fr/schedule/2019-04-23?sessionId=FVI-2317), `lazy` could cause more trouble than it helps if the object is not a really big. More info about will be published in [this I/O talk](https://events.google.com/io/schedule/events/62c1e8f8-537e-4344-87c1-003a53ebea96)

MVIMG_20190423_113310

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragment.kt [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        }
+
+        if (shouldIncludeCheckBox) {
+            addCheckbox(rootView)
+        }
+
+        return rootView
+    }
+
+    @SuppressLint("InflateParams")
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragment.kt]: Thanks :)

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragment.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonalmeida) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragment.kt]: I think you can still do that, but you don't need to use lazy init right?

File: components/feature/sitepermissions/src/main/res/layout/mozac_site_permissions_prompt.xml (hunk 1 above)
Comment (Amejia481) [reply components/feature/sitepermissions/src/main/res/layout/mozac_site_permissions_prompt.xml]: I couldn't  understand what was the issue here. I was doing some tests and it's doing the text wrapping by default.

Screenshot_20190506-220212

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragment.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Amejia481) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragment.kt]: Yes, now I'm using `get()`.

File: components/feature/sitepermissions/src/main/res/layout/mozac_site_permissions_prompt.xml (hunk 1 above)
Comment (Amejia481) [reply components/feature/sitepermissions/src/main/res/layout/mozac_site_permissions_prompt.xml]: I will merge this if there is any issue pending please feel free to open an issue and I will address it in a follow-up PR.

**Pull request merged**